internal/safari_webdriver.py

```diff
@@ -0,0 +1,1144 @@
+            "shavar.services.mozilla.com",
+            "snippets.cdn.mozilla.net",
+            "content-signature-2.cdn.mozilla.net"]
+
+    def prepare(self, job, task):
+        """Prepare the profile/OS for the browser"""
+        self.log_pos = {}
+        self.page = {}
+        self.requests = {}
+        self.main_request_headers = None
+        DesktopBrowser.prepare(self, job, task)
+        profile_template = os.path.join(os.path.abspath(os.path.dirname(__file__)),
+                                        'support', 'Safari', 'profile')
+        if not task['cached'] and os.path.isdir(profile_template):
+            try:
+                if os.path.isdir(task['profile']):
+                    shutil.rmtree(task['profile'])
+                shutil.copytree(profile_template, task['profile'])
+            except Exception:
+                logging.error('Error copying Safari profile')
+        # Delete any unsent crash reports
+        crash_dir = None
+        # Only support darwin, macOS platform
+        crash_dir = os.path.join(os.path.expanduser('~'), 'Library', 'Logs', 'CrashReporter')
+        if crash_dir and os.path.isdir(crash_dir):
+            logging.debug('Clearing crash reports in %s', crash_dir)
+            try:
+                shutil.rmtree(crash_dir)
+            except Exception:
+                pass
+
+        # Prepare the config for the extension to query
+        if self.job['message_server'] is not None:
+            config = None
+            names = ['block',
+                     'block_domains',
+                     'block_domains_except',
+                     'headers',
+                     'cookies',
+                     'overrideHosts']
+            for name in names:
+                if name in task and task[name]:
+                    if config is None:
+                        config = {}
+                    config[name] = task[name]
+            self.job['message_server'].config = config
+
+    def disable_fsync(self, command_line):
+        """Use eatmydata, on macOS, to disable fsync"""
+        try:
+            cmd = ['eatmydata', 'data']
+            logging.debug(' '.join(cmd))
+            subprocess.check_call(cmd)
+            command_line = 'eatmydata ' + command_line
+        except Exception:
+            pass
+        return command_line
+
+    def start_safari(self, job, task):
+        """Start Safari using webdriver"""
+        args = ['-profile', '"{0}"'.format(task['profile']),
+                'about:blank']
+        if self.path.find(' ') > -1:
+            command_line = '"{0}"'.format(self.path)
+        else:
+            command_line = self.path
+        command_line += ' ' + ' '.join(args)
+        command_line = self.disable_fsync(command_line)
+        from selenium import webdriver
+        capabilities = webdriver.DesiredCapabilities.SAFARI.copy()
+        self.webdriver = webdriver.Safari(desired_capabilities=capabilities)
```

Comment:
Is the extension installed at any point or is it manually installed in the system globally?

Reply:
That extension. I was dev signed and install manually. I want to get it official signed by apple, and then I could figure out how to install it via script. Right now, because it is dev signed, I have to manually install. So I did not bother to figure that out yet.